src/index.js

```diff
@@ -56,52 +59,71 @@ export default {
     exportData: true,
     snapshot: true
   },
-  paint: async function ($element, layout) {
-    const dataCube = await initializeDataCube(this, layout);
-    const designList = await initializeDesignList(this, layout);
-    const state = await initializeStore({
-      $element,
-      component: this,
-      dataCube,
-      designList,
-      layout
-    });
+  async paint ($element, layout, requestPage) {
+    const dataCube = await initializeDataCube(this, layout, requestPage);
     const editmodeClass = this.inAnalysisState() ? '' : 'edit-mode';
+    let state, designList;
+    if (dataCube === null) {
```

Comment:
Maybe could do something like;

`let state = {
        $element,
        component: this,
        dataCube,
        designList,
        layout,
        error: false
  };

if (dataCube == null) {
  state.error = true
} else {
  designList = await initializeDesignList(this, layout);
   await initializeStore(state);
}`

Reply:
I tried this @phiols , but getting many React render failures with empty and undefined assignments. So skipping this.

Reply:
Go for it! 👍